### DISCUSSION TITLE
Packaging cri-o from releases for Debian/Ubuntu/CentOSStream/Fedora

### DISCUSSION THREAD

**Discussion by ccamacho** [cri-o/cri-o#5928] [Q&A] [answered]
Hi folks,

I'm trying to build a local package mirror with some consistent versions of podman+cri-o to replace kubic.

I'm having a hard time finding docs for creating my .deb/.rpm of cri-o, Does anyone could give me some hints about it?

Thanks.

**Comment by haircommander** [comment, marked as answer]
Neat! if you download the contents from https://build.opensuse.org/package/show/devel:kubic:libcontainers:stable:cri-o:1.24/cri-o, you can see the fedora spec file (that builds for centOS{, stream}) and in the `cri-o_1.24.0~2.tar.gz` there's the debian/ directory that describes how to build the deb.

let me know if there's anything I can help you with

**Reply by ccamacho** [reply]
This is amazing!! I'll check on that and potentially publish a solution for this local kubic replacement.

**Reply by ccamacho** [reply]
@haircommander another quick question. Would you recommend using the same .spec but with debbuild? So I can reuse it for both distros? Or is it a no-go..? Thanks!

**Reply by haircommander** [reply]
yeah that should work, I think @lsm5 is using that

**Reply by lsm5** [reply]
@ccamacho here's my new iteration of the kubic packages using debbuild: https://build.opensuse.org/project/show/devel:kubic:libcontainers:unstable 

It's not without its issues though. For example, there's currently https://github.com/actions/virtual-environments/issues/5578#issuecomment-1147474715 that I need to fix. But I did check sometime back on Ubuntu 22.04, and things did work.

**Reply by jeffgtxjava** [reply]
@haircommander Quick question if you dont mind. I noticed RHEL/CentOS has removed the support for btrfs as per [link](https://access.redhat.com/solutions/197643). 

Does cri-o centOS7 RPMs come with btrfs functionality by default??

**Reply by haircommander** [reply]
we only seem to add btrfs support if we're building on fedora

**Reply by rgordill** [reply]
@haircommander, do you know why there is no cri-o in centos stream 9?

**Reply by haircommander** [reply]
yeah, OBS doesn't yet support centos stream 9
